src/topology/homotopy/H_spaces.lean

```diff
@@ -0,0 +1,150 @@
+eq.trans (by {rw Q_right, congr, apply mul_div_cancel_left, exact two_ne_zero}) $
+  set.proj_Icc_coe zero_le_one _
+
+end unit_interval
+
+namespace path
+
+open unit_interval
+
+variables {X : Type u} [topological_space X] {x y : X}
+
+/-- This is the function analogous to the one on p. 475 of [serre1951], defining a homotopy from
+the product path `e ∧ γ` to `γ`.-/
+def delayed_refl_right (θ : I) (γ : path x y) : path x y :=
+{ to_fun := λ t, γ (Q_right (t, θ)),
+  continuous_to_fun := γ.continuous.comp (continuous_Q_right.comp $ continuous.prod.mk_left θ),
+  source' := by { dsimp only, rw [Q_right_zero_left, γ.source] },
+  target' := by { dsimp only, rw [Q_right_one_left, γ.target] } }
+
+lemma continuous_delayed_refl_right : continuous (λ p : I × path x y, delayed_refl_right p.1 p.2) :=
+continuous_uncurry_iff.mp $ (continuous_snd.comp continuous_fst).path_eval $
+  continuous_Q_right.comp $ continuous_snd.prod_mk $ continuous_fst.comp continuous_fst
+
+lemma delayed_refl_right_zero (γ : path x y) : delayed_refl_right 0 γ = γ.trans (path.refl y) :=
+begin
+  ext t,
+  simp only [delayed_refl_right,
+    trans_apply, refl_extend, path.coe_mk, function.comp_app, refl_apply],
+  split_ifs, swap, conv_rhs { rw ← γ.target },
+  all_goals { apply congr_arg γ, ext1, rw Q_right_zero_right },
+  exacts [if_neg h, if_pos h],
+end
+
+lemma delayed_refl_right_one (γ : path x y) : delayed_refl_right 1 γ = γ :=
+by { ext t, exact congr_arg γ (Q_right_one_right t) }
+
+/-- This is the function on p. 475 of [serre1951]], defining a homotopy from a path `γ` to the
+product path `e ∧ γ`.-/
+def delayed_refl_left (θ : I) (γ : path x y) : path x y := (delayed_refl_right θ γ.symm).symm
+
+lemma continuous_delayed_refl_left : continuous (λ p : I × path x y, delayed_refl_left p.1 p.2) :=
+path.continuous_symm.comp $ continuous_delayed_refl_right.comp $ continuous_fst.prod_mk $
+  path.continuous_symm.comp continuous_snd
+
+lemma delayed_refl_left_zero (γ : path x y) : delayed_refl_left 0 γ = (path.refl x).trans γ :=
+by simp only [delayed_refl_left, delayed_refl_right_zero, trans_symm, refl_symm, path.symm_symm]
+
+lemma delayed_refl_left_one (γ : path x y) : delayed_refl_left 1 γ = γ :=
+by simp only [delayed_refl_left, delayed_refl_right_one, path.symm_symm]
+
+notation ` Ω_[` x `]` := path x x
+
+instance (x : X) : H_space Ω_[x] :=
+{ Hmul := ⟨λ ρ, ρ.1.trans ρ.2, continuous_trans⟩,
+  e := refl x,
+  Hmul_e_e := refl_trans_refl,
+  left_Hmul_e :=
+  { to_homotopy := ⟨⟨λ p : I × Ω_[x], delayed_refl_left p.1 p.2,
+      continuous_delayed_refl_left⟩, delayed_refl_left_zero, delayed_refl_left_one⟩,
+    prop' := by { rintro t _ (rfl : _ = _), exact ⟨refl_trans_refl.symm, rfl⟩ } },
+  right_Hmul_e :=
+  { to_homotopy := ⟨⟨λ p : I × Ω_[x], delayed_refl_right p.1 p.2,
+      continuous_delayed_refl_right⟩, delayed_refl_right_zero, delayed_refl_right_one⟩,
+    prop' := by { rintro t _ (rfl : _ = _), exact ⟨refl_trans_refl.symm, rfl⟩ } } }
```

Comment:
The homotopies seem important enough to have their own declaration? Or do you expend users to write `left_Hmul_e` for such homotopies?

Reply:
I do not know if they are really important once one knows that there is this instance, so it seems to me that keeping `left_Hmul_e` and `right_Hmul_e` (or possibly `e_Hmul` and `Hmul_e` as you suggest) is OK.